lib_eio_linux/eio_linux.ml

```diff
@@ -1053,15 +1059,17 @@ let udp_socket sock = object
         raise (Failure "Expected INET UDP socket address but got Unix domain socket address.")
 end
 
-let flow fd =
+class flow fd =
```

Comment:
Why are you changing these to a class ? Aren't they equivalent ?

Reply:
`let flow fd = ... ` defines an object which we cannot inherit. Therefore we need a `class` which defines a common ancestor class `flow`. `stream_socket` is flow + socket functionality and we inherit flow + socket capabilities here https://github.com/ocaml-multicore/eio/blob/2460bf54c0c6b4c74d8c2c887b99623f4fe84298/lib_eio_linux/eio_linux.ml#L1113-L1118

Reply:
Ah I see, TIL.